### PR TITLE
feat: Added a export csv/json button to data tables where export is available

### PR DESCRIPTION
**Pull request by Gilbert09**
## Changes
Added 2 extra buttons to the export dropdown for underlying HogQL/Person/Event queries when viewing a data table. This copies the current table view to the clipboard in either CSV or JSON format - has some special edge cases for calculated fields on the data table (such as the `*` column in events explorer).

image

## How did you test this code?
Manually tested the 3 examples; event explorer, persons, hogQL queries

### CODE REVIEW

File: frontend/src/queries/nodes/DataTable/DataTableExport.tsx [hunk 1]
```diff
@@ -39,12 +43,141 @@ function startDownload(query: DataTableNode, onlySelectedColumns: boolean): void
...
+
+        const csvData = dataTableRows.map((n) => {
+            return columns
+                .map((col, colIndex) => {
+                    if (columnDisallowList.includes(col)) {
+                        return null
+                    }
+
+                    if (col === 'person') {
+                        return n.result?.[colIndex]?.properties?.email || n.result?.[colIndex]?.uuid
```
Comment (Gilbert09) [inline frontend/src/queries/nodes/DataTable/DataTableExport.tsx:74]: Wanted to double-check whether the `uuid` was the correct alternative for anonymous users?

File: frontend/src/queries/nodes/DataTable/DataTableExport.tsx [hunk 2]
```diff
@@ -71,23 +204,59 @@ export function DataTableExport({ query }: DataTableExportProps): JSX.Element |
...
+                                    copyTableToJson(
+                                        dataTableRows,
+                                        columnsInResponse ?? columnsInQuery,
+                                        queryWithDefaults
+                                    )
+                                }
+                            }}
+                        >
+                            Copy JSON to clipboard
+                        </LemonButton>,
```
Comment (mariusandra) [inline frontend/src/queries/nodes/DataTable/DataTableExport.tsx:258]: Let's be defensive and wrap a "isPersonsNode() || isEventQuery() || ..." condition around these lines. Who knows what other nodes the export button will show up on, and having links that won't do anything will be more frustrating than not having them at all.

File: frontend/src/queries/nodes/DataTable/DataTableExport.tsx [hunk 3]
```diff
@@ -39,12 +43,141 @@ function startDownload(query: DataTableNode, onlySelectedColumns: boolean): void
...
+        return dataTableRows.map((n) => {
+            return columns.reduce((acc, col, colIndex) => {
+                if (columnDisallowList.includes(col)) {
+                    return acc
+                }
+
+                if (col === 'person') {
+                    acc[col] = n.result?.[colIndex]?.properties?.email || n.result?.[colIndex]?.uuid
+                    return acc
+                }
```
Comment (mariusandra) [inline frontend/src/queries/nodes/DataTable/DataTableExport.tsx:121]: These person checks could be unified. There's also a [function](https://github.com/PostHog/posthog/blob/8c046d74e75ebecf65e80f95fd1fa796aed23703/frontend/src/scenes/persons/person-utils.ts#L36) `asDisplay` that could be used here.

**Review by mariusandra — COMMENTED** [review]
It works! 👍  Some code suggestions inline.

**Review by mariusandra — APPROVED** [review]
Looks good 👍

**Pull request merged**